### DISCUSSION TITLE
Ultralytics HUB - Ultralytics YOLOv8 Docs

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#2870] [Docs]
# Ultralytics HUB - Ultralytics YOLOv8 Docs

Ultralytics HUB: Train & deploy YOLO models from one spot! Use drag-and-drop interface with templates & pre-training models. Check quickstart, datasets, and more.

https://docs.ultralytics.com/hub/

**Comment by GoncaloMesquita** [comment]
I am not being able to apply tracking with ultralytics and 'yolov8n.pt'. Is giving me the following error:

    results = model.track(frame, persist=True, conf=0.3, tracker='botsort.yaml', iou=0.9)
AttributeError: 'YOLO' object has no attribute 'track'
Sentry is attempting to send 2 pending events
Waiting up to 2 seconds
Press Ctrl-C to quit

**Reply by pderrenger** [reply]
@GoncaloMesquita it looks like you're encountering an issue with the tracking functionality. To ensure that you're using the correct method, please make sure you have the latest version of the Ultralytics YOLOv8 repository, as the tracking feature is a recent addition and may not be available in older versions.

If you're already using the latest version and still facing this issue, it's possible that there might be a discrepancy in the method or its usage. The `track` method should be available on the `YOLO` object if you're using the correct version.

Here's a concise example of how to use the `track` method:

```python
from ultralytics import YOLO

# Load the model
model = YOLO('yolov8n.pt')

# Perform tracking on a video source
results = model.track(source="path/to/video.mp4", conf=0.3, iou=0.9, tracker='botsort.yaml')
```

Make sure that the `tracker` parameter is pointing to the correct tracker configuration file and that the file exists in your working directory or in the predefined tracker configs directory.

If you continue to experience issues, please refer to the [Track documentation](https://docs.ultralytics.com/modes/track) for more detailed instructions and examples. If the problem persists, you may want to open an issue on the Ultralytics YOLOv8 GitHub repository with the details of the error and the steps to reproduce it. The team will be happy to assist you further. 🛠️

**Comment by Hamzajani153** [comment]
@GoncaloMesquita i also used tracking with ultralytics using results = model.track(frame, persist=True) and it give me the best output if u trying this and error also receive than try this code i'll share it might work.
import cv2
from ultralytics import YOLO

# Load the YOLOv8 model
model = YOLO('yolov8n.pt')

# Open the video file
video_path = "people.mp4"
cap = cv2.VideoCapture(video_path)

# Loop through the video frames
while cap.isOpened():
    # Read a frame from the video
    success, frame = cap.read()

    if success:
        # Run YOLOv8 tracking on the frame, persisting tracks between frames
        results = model.track(frame, persist=True)

        # Visualize the results on the frame
        annotated_frame = results[0].plot()

        # Display the annotated frame
        cv2.imshow("YOLOv8 Tracking", annotated_frame)

        # Break the loop if 'q' is pressed
        if cv2.waitKey(1) & 0xFF == ord("q"):
            break
    else:
        # Break the loop if the end of the video is reached
        break

# Release the video capture object and close the display window
cap.release()
cv2.destroyAllWindows()
